host/frontend/host-orchestrator/operator.go

```diff
@@ -0,0 +1,400 @@
+	}).Methods("POST")
+	router.HandleFunc("/polled_connections/{connId}/messages", func(w http.ResponseWriter, r *http.Request) {
+		messages(w, r, polledSet)
+	}).Methods("GET")
+	router.HandleFunc("/polled_connections", func(w http.ResponseWriter, r *http.Request) {
+		createPolledConnection(w, r, pool, polledSet)
+	}).Methods("POST")
+	router.HandleFunc("/infra_config", func(w http.ResponseWriter, r *http.Request) {
+		replyJSON(w, config)
+	}).Methods("GET")
+	fs := http.FileServer(http.Dir("static"))
+	router.PathPrefix("/").Handler(fs)
+	return router
+}
+
+// Device endpoint
+func deviceWs(w http.ResponseWriter, r *http.Request, pool *DevicePool, config interface{}) {
+	log.Println(r.URL)
+	ws := NewJsonWs(w, r)
+	if ws == nil {
+		return
+	}
+	// Serve the websocket in its own thread
+	go func() {
+		defer ws.Close()
+		var msg RegisterMsg
+		ok := ws.Recv(&msg)
+		if !ok {
+			wsReplyError(ws, "Websocket closed unexpectedly")
+			return
+		}
+		if msg.Type != "register" {
+			wsReplyError(ws, "First device message must be the registration")
+			return
+		}
+		id := msg.DeviceId
+		if id == "" {
+			wsReplyError(ws, "Missing device_id")
+			return
+		}
+		info := msg.Info
+		if info == nil {
+			log.Println("No device info provided by: ", id)
+			info = make(map[string]interface{})
+		}
+		port := msg.Port
+		device := NewDevice(ws, id, port, info)
+		if !pool.Register(device, id) {
+			wsReplyError(ws, fmt.Sprintln("Device id already taken: ", id))
+			return
+		}
+		defer pool.Unregister(id)
+		if !device.Send(config) {
+			log.Println("Failed to send config to device")
+			return
+		}
+		for {
+			var msg ForwardMsg
+			ok := ws.Recv(&msg)
+			if !ok {
+				log.Println("Device websocket closed")
+				return
+			}
+			if msg.Type != "forward" {
+				wsReplyError(ws, fmt.Sprintln("Unrecognized message type: ", msg.Type))
+				return
+			}
+			clientId := msg.ClientId
+			if clientId == 0 {
+				wsReplyError(ws, "Device forward message missing client id")
+				return
+			}
+			payload := msg.Payload
+			if payload == nil {
+				wsReplyError(ws, "Device forward message missing payload")
+				return
+			}
+			dMsg := map[string]interface{}{
+				"message_type": "device_msg",
+				"payload":      payload,
+			}
+			if !device.ToClient(clientId, dMsg) {
+				log.Println(ws, "Missing client ", int(clientId), " for device ", id)
+				wsReplyError(ws, fmt.Sprintln("Client disconnected: ", clientId))
+			}
+		}
+	}()
+}
+
+// General client endpoints
+
+func listDevices(w http.ResponseWriter, r *http.Request, pool *DevicePool) {
+	if err := replyJSON(w, pool.DeviceIds()); err != nil {
+		log.Println(err)
+	}
+}
+
+func deviceFiles(w http.ResponseWriter, r *http.Request, pool *DevicePool) {
+	vars := mux.Vars(r)
+	devId := vars["deviceId"]
+	dev := pool.GetDevice(devId)
+	if dev == nil {
+		http.NotFound(w, r)
+		return
+	}
+	path := vars["path"]
+	if shouldIntercept(path) {
+		http.ServeFile(w, r, fmt.Sprintf("intercept%s", path))
```

Comment:
`shouldIntercept` does `fmt.Sprintf("intercept/%s", ...)` while this does `fmt.Sprintf("intercept%s", ...)`, what's the distinction between these situations?

Reply:
Fixed
There is no difference, path includes the leading '/' so it's not needed in `shouldIntercept`. It worked because apparently some/path is equivalent to some//path in linux.